src/test/java/co/aurasphere/botmill/fb/demo/behavior/AnnotatedTemplatedBehaviour.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2016 BotMill.io
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package co.aurasphere.botmill.fb.demo.behavior;
+
+import java.math.BigDecimal;
+import co.aurasphere.botmill.fb.AbstractFbBot;
+import co.aurasphere.botmill.fb.autoreply.AutoReply;
+import co.aurasphere.botmill.fb.autoreply.MessageAutoReply;
+import co.aurasphere.botmill.fb.event.FbBotMillEventType;
+import co.aurasphere.botmill.fb.model.annotation.BotMillController;
+import co.aurasphere.botmill.fb.model.incoming.MessageEnvelope;
+import co.aurasphere.botmill.fb.model.outcoming.FbBotMillResponse;
+import co.aurasphere.botmill.fb.model.outcoming.factory.ButtonFactory;
+import co.aurasphere.botmill.fb.model.outcoming.factory.ReplyFactory;
+
+/**
+ * The Class TemplateBehavior.
+ */
+public class AnnotatedTemplatedBehaviour extends AbstractFbBot {
+	
+	@BotMillController(eventType=FbBotMillEventType.MESSAGE, text="text message", caseSensitive = true)
+	public void replyText() {
+		reply(new MessageAutoReply("simple text message"));
+	}
+	
```

Comment:
I like this a lot, but I was wondering if it would be better to directly return a reply from the method instead of calling the method reply. What do you think?